### ISSUE TITLE
[Bug] r.in.pdal: compilation error with c++14 and PDAL 2.5.0

### ISSUE THREAD

**Issue by neteler** [OSGeo/grass#2750]
**Describe the bug**
Recently, PDAL 2.5.0 has been published. Building it on Fedora 38 triggers a compilation error in GRASS GIS 8.2.git:

`r.in.pdal/info.cpp:103: undefined reference to `pdal::operator<<(std::ostream&, pdal::LasHeader const&)'`

**To Reproduce**

- using Fedora 38 docker image
- download G8.2 tarball
- compile G8.2, using  c++14 patch (https://src.fedoraproject.org/rpms/grass/blob/rawhide/f/grass-pdal.patch)
- see error:

```
[r.in.pdal]# make
: && g++ -L/root/grass-8.2.0/dist.x86_64-pc-linux-gnu/lib -L/root/grass-8.2.0/dist.x86_64-pc-linux-gnu/lib -Wl,--export-dynamic -Wl,-rpath-link,/root/grass-8.2.0/dist.x86_64-pc-linux-gnu/lib -Wl,-rpath,/usr/local/grass82/lib -o /root/grass-8.2.0/dist.x86_64-pc-linux-gnu/bin/r.in.pdal OBJ.x86_64-pc-linux-gnu/bin_update.o OBJ.x86_64-pc-linux-gnu/bin_write.o OBJ.x86_64-pc-linux-gnu/filters.o OBJ.x86_64-pc-linux-gnu/lidar.o OBJ.x86_64-pc-linux-gnu/point_binning.o OBJ.x86_64-pc-linux-gnu/projection.o OBJ.x86_64-pc-linux-gnu/rast_segment.o OBJ.x86_64-pc-linux-gnu/string_list.o OBJ.x86_64-pc-linux-gnu/grasslidarfilter.o OBJ.x86_64-pc-linux-gnu/info.o OBJ.x86_64-pc-linux-gnu/main.o    -lgrass_raster.8.2 -lgrass_segment.8.2 -lgrass_gproj.8.2 -lgrass_vector.8.2 -lgrass_dbmiclient.8.2 -lgrass_dbmibase.8.2  -lgrass_gis.8.2 -lm  -L/usr/lib -lpdalcpp -lgrass_gmath.8.2   -lm 
/usr/bin/ld: /tmp/ccOrqoOT.ltrans0.ltrans.o: in function `print_lasinfo(StringList*)':
/root/grass-8.2.0/raster/r.in.pdal/info.cpp:103: undefined reference to `pdal::operator<<(std::ostream&, pdal::LasHeader const&)'
collect2: error: ld returned 1 exit status
make: *** [../../include/Make/Module.make:18: /root/grass-8.2.0/dist.x86_64-pc-linux-gnu/bin/r.in.pdal] Error 1
```

**Expected behavior**
No compilation error :-)

**System description (please complete the following information):**

- Operating System: Fedora 38
- GRASS GIS version: 8.2.x

**Comment by neteler** [comment]
More detailed error message:
```

g++  -g -O2  -I/root/grass-8.2.0/dist.x86_64-pc-linux-gnu/include -I/root/grass-8.2.0/dist.x86_64-pc-linux-gnu/include   -I/usr/include/gdal -I/usr/include -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -U_FORTIFY_SOURCE -Wp,-U_FORTIFY_SOURCE -Wp,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -std=c++11 -std=c++14 -DPACKAGE=\""grassmods"\"    -I/usr/include -I/usr/include/gdal -I/usr/include/libxml2 -I/root/grass-8.2.0/dist.x86_64-pc-linux-gnu/include -I/root/grass-8.2.0/dist.x86_64-pc-linux-gnu/include -DRELDIR=\"raster/r.in.pdal\" -o OBJ.x86_64-pc-linux-gnu/info.o -c info.cpp
info.cpp: In function ‘void print_lasinfo(StringList*)’:
info.cpp:103:22: warning: ‘std::ostream& pdal::operator<<(std::ostream&, const LasHeader&)’ is deprecated [-Wdeprecated-declarations]
  103 |         std::cout << las_header;
      |                      ^~~~~~~~~~
In file included from info.h:20,
                 from info.cpp:12:
/usr/include/pdal/io/LasHeader.hpp:288:26: note: declared here
  288 |     friend std::ostream& operator<<(std::ostream& ostr, const LasHeader& h);
      |                          ^~~~~~~~
g++  -g -O2  -I/root/grass-8.2.0/dist.x86_64-pc-linux-gnu/include -I/root/grass-8.2.0/dist.x86_64-pc-linux-gnu/include   -I/usr/include/gdal -I/usr/include -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -U_FORTIFY_SOURCE -Wp,-U_FORTIFY_SOURCE -Wp,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -std=c++11 -std=c++14 -DPACKAGE=\""grassmods"\"    -I/usr/include -I/usr/include/gdal -I/usr/include/libxml2 -I/root/grass-8.2.0/dist.x86_64-pc-linux-gnu/include -I/root/grass-8.2.0/dist.x86_64-pc-linux-gnu/include -DRELDIR=\"raster/r.in.pdal\" -o OBJ.x86_64-pc-linux-gnu/main.o -c main.cpp
: && g++ -L/root/grass-8.2.0/dist.x86_64-pc-linux-gnu/lib -L/root/grass-8.2.0/dist.x86_64-pc-linux-gnu/lib -Wl,--export-dynamic -Wl,-rpath-link,/root/grass-8.2.0/dist.x86_64-pc-linux-gnu/lib -Wl,-rpath,/usr/local/grass82/lib -o /root/grass-8.2.0/dist.x86_64-pc-linux-gnu/bin/r.in.pdal OBJ.x86_64-pc-linux-gnu/bin_update.o OBJ.x86_64-pc-linux-gnu/bin_write.o OBJ.x86_64-pc-linux-gnu/filters.o OBJ.x86_64-pc-linux-gnu/lidar.o OBJ.x86_64-pc-linux-gnu/point_binning.o OBJ.x86_64-pc-linux-gnu/projection.o OBJ.x86_64-pc-linux-gnu/rast_segment.o OBJ.x86_64-pc-linux-gnu/string_list.o OBJ.x86_64-pc-linux-gnu/grasslidarfilter.o OBJ.x86_64-pc-linux-gnu/info.o OBJ.x86_64-pc-linux-gnu/main.o    -lgrass_raster.8.2 -lgrass_segment.8.2 -lgrass_gproj.8.2 -lgrass_vector.8.2 -lgrass_dbmiclient.8.2 -lgrass_dbmibase.8.2  -lgrass_gis.8.2 -lm  -L/usr/lib -lpdalcpp -lgrass_gmath.8.2   -lm 
/usr/bin/ld: /tmp/ccM5Tc21.ltrans0.ltrans.o: in function `print_lasinfo(StringList*)':
/root/grass-8.2.0/raster/r.in.pdal/info.cpp:103: undefined reference to `pdal::operator<<(std::ostream&, pdal::LasHeader const&)'
collect2: error: ld returned 1 exit status
make: *** [../../include/Make/Module.make:18: /root/grass-8.2.0/dist.x86_64-pc-linux-gnu/bin/r.in.pdal] Error 1

```

**Comment by nilason** [comment]
> #2749 will probably solve this.

And no need for the patch.

**Comment by neteler** [comment]
The proposed PR #2749 fixes the issue. Like that no hardcoded C++ standard is used.

On Fedora 38 it corresponds to:

```
g++ -dM -E -x c++  /dev/null | grep -F __cplusplus
#define __cplusplus 201703L
```

which is C++ Standard `C++17`.

@manisandro: we'll then backport PR #2749 to the upcoming G8.2.1 version, so the new [grass-pdal.patch](https://src.fedoraproject.org/rpms/grass/blob/rawhide/f/grass-pdal.patch) should no longer be needed. And G8.2.1 is basically due "now".

**Comment by neteler** [comment]
Fixed in main (cb8293e) and backported to 8.2 (2f20bc1).
Will be part of the upcoming GRASS GIS 8.2.1.

**Comment by neteler** [comment]
BTW: GRASS GIS 8.2.1 now successfully packaged for Fedora with PDAL 2.5.0 support:
https://src.fedoraproject.org/rpms/grass
